lib/ecl/ecl_grav.cpp

```diff
@@ -530,29 +521,25 @@ static ecl_grav_survey_type * ecl_grav_survey_alloc_RFIP(ecl_grav_type * ecl_gra
 static void ecl_grav_survey_free( ecl_grav_survey_type * grav_survey ) {
   free( grav_survey->name );
   free( grav_survey->porv );
-  vector_free( grav_survey->phase_list );
-  hash_free( grav_survey->phase_map );
-  free( grav_survey );
-}
+  for (auto * phase : grav_survey->phase_list)
+    ecl_grav_phase_free( phase );
 
-static void ecl_grav_survey_free__( void * __grav_survey ) {
-  ecl_grav_survey_type * grav_survey = ecl_grav_survey_safe_cast( __grav_survey );
-  ecl_grav_survey_free( grav_survey );
+  delete grav_survey;
 }
 
 
 
+
 static double ecl_grav_survey_eval( const ecl_grav_survey_type * base_survey,
                                     const ecl_grav_survey_type * monitor_survey ,
                                     ecl_region_type * region ,
                                     double utm_x , double utm_y , double depth, int phase_mask) {
-  int phase_nr;
   double deltag = 0;
-  for (phase_nr = 0; phase_nr < vector_get_size( base_survey->phase_list ); phase_nr++) {
-    ecl_grav_phase_type * base_phase    = (ecl_grav_phase_type*)vector_iget( base_survey->phase_list , phase_nr );
+  for (int phase_nr = 0; phase_nr < static_cast<int>(base_survey->phase_list.size()); phase_nr++) {
```

Comment:
Should this not be the other way around, changing the type of `phase_nr` instead?

Reply:
Yes agree - fixed.